src/main/frontend/handler/common/file.cljs

```diff
@@ -41,25 +49,62 @@
                               :clear? false}]))))))
 
 (defn- validate-and-get-blocks-to-delete
+  "An implementation for the delete-blocks-fn in graph-parser/parse-file"
   [repo-url db file-page file-path retain-uuid-blocks]
   (validate-existing-file repo-url file-page file-path)
   (graph-parser/get-blocks-to-delete db file-page file-path retain-uuid-blocks))
 
+(defn- diff-merge-uuids
+  "Infer new uuids from existing DB data and diff with the new AST
+   Return a list of uuids for the new blocks"
+  [format ast content {:keys [page-name] :as options}]
+  (let [base-diffblocks (diff-merge/db->diff-blocks page-name)
+        income-diffblocks (diff-merge/ast->diff-blocks ast content format options)
+        diff-ops (diff-merge/diff base-diffblocks income-diffblocks)
+        new-uuids (diff-merge/attachUUID diff-ops (map :uuid base-diffblocks))]
+    (bean/->clj new-uuids)))
+
+(defn- reset-file!-impl
```

Comment:
Another common convention for an inner fn is `NAME*` e.g. `reset-file!*`. May read nicer